File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/network/method/MethodTest.java
package com.github.dedis.popstellar.model.network.method;

import org.junit.Test;

import static org.junit.Assert.*;

public class MethodTest {

  @Test
  public void getDataClassTest() {
    assertEquals(Subscribe.class, Method.SUBSCRIBE.dataClass);
    assertEquals(Unsubscribe.class, Method.UNSUBSCRIBE.dataClass);
    assertEquals(Publish.class, Method.PUBLISH.dataClass);
    assertEquals(Broadcast.class, Method.MESSAGE.dataClass);
    assertEquals(Catchup.class, Method.CATCHUP.dataClass);
  }

  @Test
  public void expectResultTest() {
    assertTrue(Method.SUBSCRIBE.expectResult());
    assertTrue(Method.UNSUBSCRIBE.expectResult());
    assertTrue(Method.PUBLISH.expectResult());
    assertFalse(Method.MESSAGE.expectResult());
    assertTrue(Method.CATCHUP.expectResult());
  }

  @Test
  public void findTest() {
    assertNull(Method.find("not a method"));
    assertEquals(Method.SUBSCRIBE, Method.find("subscribe"));
  }
}

File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/network/method/MethodTest.kt
package com.github.dedis.popstellar.model.network.method

import com.github.dedis.popstellar.model.network.method.Method.Companion.find
import org.junit.Assert
import org.junit.Test

class MethodTest {
  @Test
  fun dataClassTest() {
    Assert.assertEquals(Subscribe::class.java, Method.SUBSCRIBE.dataClass)
    Assert.assertEquals(Unsubscribe::class.java, Method.UNSUBSCRIBE.dataClass)
    Assert.assertEquals(Publish::class.java, Method.PUBLISH.dataClass)
    Assert.assertEquals(Broadcast::class.java, Method.MESSAGE.dataClass)
    Assert.assertEquals(Catchup::class.java, Method.CATCHUP.dataClass)
  }

  @Test
  fun expectResultTest() {
    Assert.assertTrue(Method.SUBSCRIBE.expectResult())
    Assert.assertTrue(Method.UNSUBSCRIBE.expectResult())
    Assert.assertTrue(Method.PUBLISH.expectResult())
    Assert.assertFalse(Method.MESSAGE.expectResult())
    Assert.assertTrue(Method.CATCHUP.expectResult())
  }

  @Test
  fun findTest() {
    Assert.assertNull(find("not a method"))
    Assert.assertEquals(Method.SUBSCRIBE, find("subscribe"))
  }
}
